DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/DelegateCourses/DelegateCoursesViewModel.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateCourses
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models.Courses;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup;
+
+    public class DelegateCoursesViewModel : BaseSearchablePageViewModel
+    {
+        public DelegateCoursesViewModel(
+            IEnumerable<CourseStatisticsWithAdminFieldResponseCounts> courses,
+            IEnumerable<string> categories,
+            IEnumerable<string> topics,
+            string? searchString,
+            string sortBy,
+            string sortDirection,
+            string? filterBy,
+            int page,
+            int? itemsPerPage
+        ) : base(
+            searchString,
+            page,
+            true,
+            sortBy,
+            sortDirection,
+            filterBy,
+            itemsPerPage ?? DefaultItemsPerPage,
+            "Search courses"
+        )
+        {
+            var sortedItems = GenericSortingHelper.SortAllItems(
+                courses.AsQueryable(),
+                sortBy,
+                sortDirection
+            );
+            var filteredItems = FilteringHelper.FilterItems(sortedItems.AsQueryable(), filterBy).ToList();
+            var searchedItems = GenericSearchHelper.SearchItems(filteredItems, SearchString).ToList();
+            var paginatedItems = GetItemsOnCurrentPage(searchedItems);
+
+            MatchingSearchResults = searchedItems.Count;
+            SetTotalPages();
+            Courses = paginatedItems.Select(c => new SearchableCourseStatisticsViewModel(c));
+            Filters = CourseStatisticsViewModelFilterOptions.GetFilterOptions(categories, topics);
```

Comment:
@SteveJacksonSoft We are going to be displaying all the filterable tags because of this, but the ticket mentions only showing the Active tag. Why do we only want to show the Active tag? I think we show all filterable tags everywhere else, including the Course Setup cards that these should be pretty much identical to.

Reply:
Oh, good catch. I checked with Kevin whether he wanted the `Visible in Learning Portal` tag and he said no because it's not relevant here: https://softwire.slack.com/archives/C016Q1GR2G5/p1644393276194899?thread_ts=1644339810.552289&cid=C016Q1GR2G5
Though that could have been under the assumption that it would save time not to show it.